archive/d/dart/factorial.dart

```diff
@@ -0,0 +1,20 @@
+// Issue 4969
+void main(List<String> args){
+  const String error_message = "Usage: please input a non-negative integer";
+  try{
+
+    int factorial = 1;
+    int num_needed = int.parse(args[0]);
+    if (num_neede.isNegative == true){
+      print();
+    }
+    else{
+      for (int i_index = 1; i_index <= num_needed; i_index ++){
+        factorial *= i_index;
+      }
+    }
```

Comment:
The factorial value needs to be displayed. Please consult the [Project Description](https://sampleprograms.io/projects/factorial/)